app/models/maestrano/connector/rails/services/data_consolidator.rb

```diff
@@ -0,0 +1,153 @@
+module Maestrano::Connector::Rails::Services
+  class DataConsolidator
+    def initialize(organization, entity_self, options)
+      @is_a_subentity = entity_self.class.included_modules.include?(Maestrano::Connector::Rails::Concerns::SubEntityBase)
+      @organization = organization
+      @current_entity = entity_self
+      @opts = options
+    end
+
+    def consolidate_singleton(connec_entities, external_entities)
+      return {connec_entities: [], external_entities: []} if external_entities.empty? && connec_entities.empty?
+
+      idmap = @current_entity.class.find_or_create_idmap(organization_id: @organization.id)
+      # No to_connec, to_external and inactive consideration here as we don't expect those workflow for singleton
+
+      keep_external = if external_entities.empty?
```

Comment:
Maybe add a comment to explain what keep_external is gonna do?
It's not obvious for a connector noob like me :)